### PR TITLE
Add italian locale

### PR DESCRIPTION
**Pull request by albertorestifo**
I've put my nationality to good use and created an Italian locale.

I based it off the Spanish locale files, given how similar the two languages are.

### CODE REVIEW

File: src/locale/it/build_distance_in_words_locale/test.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
+1 @flow
```
Comment (LeshaKoss) [inline src/locale/it/build_distance_in_words_locale/test.js]: @	albertorestifo please fix this
`// @flow`

Comment (albertorestifo) [inline src/locale/it/build_distance_in_words_locale/test.js]: 👍

**Comment by LeshaKoss** [comment]
@albertorestifo tests are falling 🤔

**Comment by albertorestifo** [comment]
Sorry, I forgot to run the tests after I did a change.
Now they should pass :bowtie:

**Comment by albertorestifo** [comment]
Please note that in Italian for single values, we don't use the number.

> 1 minute ago

Becomes

> un minuto fa

I've checked with Moment.js Italian locale and they use this same behaviour.

**Comment by albertorestifo** [comment]
Something that maybe we can discuss:

In Italian

> half a minute ago

translates to

> mezzo minuto fa

however I've never said it. What if instead I translate "half a minute" to a more commonly used expression:

> some seconds ago

> alcuni secondi fa

It's more vague for sure, but it's more commonly used

**Comment by kossnocorp** [comment]
:+1: for commonly used form!

**Comment by LeshaKoss** [comment]
@albertorestifo yes, let's use the common form

File: src/locale/it/build_format_locale/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    formattingTokensRegExp: buildFormattingTokensRegExp(formatters)
+  }
+}
+
+function ordinal (number) {
+  var rem100 = number % 100
+  if (rem100 > 20 || rem100 < 10) {
+    switch (rem100 % 10) {
+      case 1:
+        return number + 'º'
```
Comment (LeshaKoss) [inline src/locale/it/build_format_locale/index.js]: @albertorestifo I think
```javascript
function ordinal (number) {
  return number + 'º'
}
```
is enough

(from https://github.com/date-fns/date-fns/pull/297#discussion_r91238275)

**Comment by albertorestifo** [comment]
There we go guys, applied all the changes 🚀

**Comment by LeshaKoss** [comment]
@albertorestifo thank you! Italian locale was released with [1.16.0](https://github.com/date-fns/date-fns/releases/tag/v1.16.0) (diff: https://github.com/date-fns/date-fns/compare/v1.15.1...v1.16.0)
🇮🇹🇸🇲🇨🇭🇻🇦

**Pull request merged**